### ISSUE TITLE
[FEATURE] Make the repository compatible with the upcoming external_components feature

### ISSUE THREAD

**Issue by mmakaay** [mmakaay/esphome-xiaomi_bslamp2#16]
The ESPHome team is working on a great option to work with components that are hosted on GitHub.
This makes it very easy to use third party components, without all the hassle of downloading the code, and manually moving it to the correct directories on the build system. For example the Bedside Lamp 2 code could be included in the build by adding something like this to the yaml config:

```yaml
external_components:
  - source:
      type: git
      url: https://github.com/mmakaay/esphome-xiaomi_bslamp2
      ref: 1.0.0-RC1
```

To make this work, the repository needs to be structured in a standard way.
Because of the big jump in usability, I will happily make my repo compatible with the requirements.

The change is not very complicated:

* Create a directory `components/xiaomi_bslamp2`
* Move all code that is required for the component to work into that new directory

I will do this, before going to a first stable release of the code.

**Comment by mmakaay** [comment]
I created a branch in which this change was done. Using that I was able to test the new external components feature. It works very well. Way better than having to explain users of my code that they should move files in place manually for this to work!

The concept is proven, but I will not yet modify the repository.
When I do so, then the current installation method of adding the repo to `config/custom_components` will be less practical to do (since the actual component code will be in a subdirectory and a simple git clone will not be feasible anymore for installing the code as a custom component).

Thus, this will be implemented, but only after external components are available from the stable version of ESPHome.
